### ISSUE TITLE
Renaming proposition.

### ISSUE THREAD

**Issue by kaligrafy** [chairemobilite/evolution#745]
# Proposition for renaming classes, attributes and actions, plus users categories:

### Interview manipulation by use case:

- Validation: Live validation of interviews data, shown to respondent and interviewers during the interview
- Correction: Manual correction of interviews data, done by admin/correctors
- Audit: Automated validation of interviews data, done by the server and shown to admin/correctors

### Users:

- Admin: Users with the admin role can administer the surveys, including monitoring and correcting interviews
- Corrector: Users with the correctors role can correct interviews data
- Participant: Users with the participant role can fill the surveys (also called Respondent)
- Interviewer: Users with the interviewer role can fill out interviews for a participant by phone, in person or remotely, or can assist participants with their interviews

Please discuss if you agree, and if not, propose better (If you don't like a name, propose a better one :-) )

**Comment by samuel-duhaime** [comment]
For the admin user, do we want to put some role, because right now admin is too broad of a type ?

**Comment by tahini** [comment]
> For the admin user, do we want to put some role, because right now admin is too broad of a type ?

Yes we do, eventually, none of us should be admin, but we should all have permissions of some kind. We have all the framework for this and we have interviewer/validator roles, but we stopped there as it was enough at the time and didn't get time to work on it...

**Comment by tahini** [comment]
Sounds good for the renaming. But is that exhaustive enough? What about the case of somebody visualizing the audits (in the `summary` page) and just setting the interview as valid/invalid, complete/incomplete, questionable? This process does not require correction and we may not want all users who can do that to actually have the permission to correct the interview?

You think of OD surveys, but there are many others where we may just want to manually [in]validate them without giving right to edit.

**Comment by kaligrafy** [comment]
I think we can still do better renaming and separating users/permissions

**Comment by tahini** [comment]
Roles/permissions and renaming are 2 separate things anyway.

**Comment by kaligrafy** [comment]
@tahini Waiting for our review :-)

**Comment by tahini** [comment]
@kaligrafy sounds good. I already used this proposition in the ISCTSC article!

**Comment by tahini** [comment]
For eventual PRs doing the renaming, please do one at a time to make it simpler to track, as each may impact many variable names and quite a burden to review ;-)

**Comment by kaligrafy** [comment]
## Nomenclature updated in the code:
* [ ] Answer
* [ ] Response
* [ ] RespondentInterview
* [ ] AssistedInterview
* [ ] RealTimeValidation
* [ ] Audit
* [ ] AuditCheck
* [ ] AuditCheckResult
* [ ] AuditReport
* [ ] Review
* [ ] Correction
* [ ] Enhancement
* [ ] Monitoring
* [ ] Export
* [ ] Respondent
* [ ] Interviewer
* [ ] SystemAuditor
* [ ] Reviewer
* [ ] Corrector
* [ ] Admin

**Comment by samuel-duhaime** [comment]
@kaligrafy Can you add the remaining tasks for the renaming?

**Comment by kaligrafy** [comment]
It will take a lot more time to rename everything. I will add tasks later on, but this will be for next year's surveys,
